### ISSUE TITLE
Console output is missing global variables (HeishaMon v5, firmware 3.5)

### ISSUE THREAD

**Issue by blb4github** [heishamon/HeishaMon#486]
In the 'old' HeishaMon version the global variable where always visible on the console, but now on HeishaMon v5 with firmware 3.5 I don't see the global variables in the console output.
<img width="591" alt="Scherm­afbeelding 2024-05-21 om 23 07 44" src="https://github.com/Egyras/HeishaMon/assets/35904565/4e828950-4b48-4bfe-b532-17ec3cd6e417">

I do see them via serial logging:

```
909372: ==== Operating_Mode_State ====
909372: rule #12 was executed in 267 microseconds
909372: 
>>> local variables

909373: 
>>> global variables

 4 #allowQuietMode = 1
 5 #allowSyncHP = 1
 6 #allowSyncOT = 1
 7 #maxTa = 36
 8 #C3WayValve = 0
 9 #CCompressorFrequency = 0
10 #chEnable = 0
11 #chEnableOffTime = 0
12 #chEnableTimeOff = 5577
13 #CompressorRunTime = -14
14 #CompressorStateChangeTime = 5576
15 #CompressorState = 0
16 #CQuietMode = 3
17 #CDeFrost = 0
18 #DHWRun = -1
19 #FirstBoot = 2
20 #RHPState = 0
21 #RMaxPumpDuty = 85
22 #COperatingMode = 0
23 #OutsideTemp = 17
24 #PreviousHPS = -1
25 #PreviousOM = -1
26 #RemoteOverRide = 0
27 #RQuietMode = 3
28 #SHifT = -5
29 #TCompressorFreq = 24
30 #RoomTempDelta = 3.09766
31 #time = 5590
32 #ST = -64
```

**Comment by stumbaumr** [comment]
Should be solved in this build https://github.com/IgorYbema/HeishaMon/actions/runs/9553126605.

**Comment by Binifada1956** [comment]
Heishamon V5, SW Alpha-e17aac1

I've tested it. Sorry, still no Variables

![grafik](https://github.com/Egyras/HeishaMon/assets/99736262/204d742a-1c7d-4603-8bb3-0a4bb935ff3e)

**Comment by stumbaumr** [comment]
Mmmh, very strange, I do get global variables with version Alpha-e17aac1

![grafik](https://github.com/Egyras/HeishaMon/assets/12425441/6c89d546-c5a6-4972-8887-219e1bbfdc30)

![grafik](https://github.com/Egyras/HeishaMon/assets/12425441/d2849f89-45b5-4117-b48a-3d077d37bb3b)

**Comment by stumbaumr** [comment]
Maybe there is a difference regarding console logging in the ESP32 vs. the ESP8266 builds.

**Comment by Binifada1956** [comment]
Do you have heishamon V4 or V5?

**Comment by stumbaumr** [comment]
V4 - ESP8266 based. So the rules are still able to produce variable output - I think it is the build...

**Comment by Binifada1956** [comment]
The rules worked with the "old" V4 Heishamon and the variables were output correctly.
The rules also run in V5, only the output in the console remains empty. Therefore no control over the content of the variables is possible.

**Comment by blb4github** [comment]
I can confirm the variables are not visible in console output on V5, fw Alpha-e17aac1. I do see them in serial logging.

**Comment by stumbaumr** [comment]
This is probably something @IgorYbema should know about... somehow on ESP32 variables do not show in console output - but on ESP8266 they do...

**Comment by Binifada1956** [comment]
May be @CurlyMoo also …?

**Comment by CurlyMoo** [comment]
For ESP32 you should consult @IgorYbema

**Comment by IgorYbema** [comment]
This commit should fix it: https://github.com/IgorYbema/HeishaMon/commit/48e315a51e26f4eb8fb1d45a534adc28595700c1
Tested it ok

**Comment by blb4github** [comment]
I've tested it with my rules set I mentioned in https://github.com/Egyras/HeishaMon/issues/503, it's not working for that rules set. I have figured out it's working without the 'print' commands (
[Heishamon_rules20240819_without_print.txt](https://github.com/user-attachments/files/16681082/Heishamon_rules20240819_without_print.txt)), thanks, great!!

**Comment by McMagellan** [comment]
I've had similar problems with my ruleset.
Version #544 contains the most developed rules version. It includes SetCurves, Concat, Print. String handling was implemented there and a few trigger detail improvements were made. 
Unfortunately, there are memory conflicts that can lead to a crash when a new web server session is started. I'm living with it at the moment. Curlymoo is working on it.
Although my ruleset could be loaded (with the save button), there is a second loading process after a reset that reloads the rules. With this, Heishamon always crashes.
My non-binding tip: I don't have these crashes anymore since I divided the longest rule, which was 640 bytes long, into 3 rules.
For you, rule number 2 is also very long. Try that out. What happens if you simply leave out rule number 2 and the print commands? But I still have Heishamon V4.
